src/hashkit/nc_ketama.c

```diff
@@ -25,7 +25,7 @@
 
 #define KETAMA_CONTINUUM_ADDITION   10  /* # extra slots to build into continuum */
 #define KETAMA_POINTS_PER_SERVER    160 /* 40 points per hash */
-#define KETAMA_MAX_HOSTLEN          86
+#define KETAMA_MAX_HOSTLEN          273
```

Comment:
add commeent `/* max domain length(256bytes)+port(5bytes)+weight(11bytes) */`

Reply:
fix 
/* 273 is 255(domain or ip)+1(:)+5(port)+1(-)+10(uint32)+1(\0) */